src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -0,0 +1,236 @@
+package org.corfudb.infrastructure;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.runtime.view.LayoutView;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+
+/**
+ * Instantiates and performs failure detection adn handling asynchronously.
+ * <p>
+ * Created by zlokhandwala on 9/28/16.
+ */
+@Slf4j
+public class ManagementServer extends AbstractServer {
+
+    @Override
+    public void reset() {
+    }
+
+    @Override
+    public void reboot() {
+    }
+
+    /**
+     * The options map.
+     */
+    private Map<String, Object> opts;
+    private ServerContext serverContext;
+
+    private IFailureDetectorPolicy failureDetectorPolicy;
```

Comment:
You are right but if there is a transient failure, this race condition does not cause an issue.
If the node has actually failed, the failure will certainly show up in a few pings.

If I correct this race by a blocking call, won't it will slow down the detection? (Consider round-robbin poll of 1000 nodes.)